ndc-models/src/lib.rs

```diff
@@ -420,11 +444,128 @@ pub struct Query {
     pub limit: Option<u32>,
     /// Optionally offset from the Nth result
     pub offset: Option<u32>,
+    /// Optionally specify how rows should be ordered
     pub order_by: Option<OrderBy>,
+    /// Optionally specify a predicate to apply to the rows
     pub predicate: Option<Expression>,
+    /// Optionally group and aggregate the selected rows
+    pub groups: Option<Grouping>,
 }
 // ANCHOR_END: Query
 
+// ANCHOR: Grouping
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Grouping")]
+pub struct Grouping {
+    /// Dimensions along which to partition the data
+    pub dimensions: Vec<Dimension>,
+    /// Aggregates to compute in each group
+    pub aggregates: IndexMap<String, Aggregate>,
+    /// Optionally specify a predicate to apply after grouping rows
+    pub predicate: Option<GroupExpression>,
+    /// Optionally specify how rows should be ordered
+    pub order_by: Option<GroupOrderBy>,
+    /// Optionally limit to N groups
+    pub limit: Option<u32>,
+    /// Optionally offset from the Nth group
+    pub offset: Option<u32>,
+}
+// ANCHOR_END: Grouping
+
+// ANCHOR: GroupExpression
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Group Expression")]
+pub enum GroupExpression {
+    And {
+        expressions: Vec<GroupExpression>,
+    },
+    Or {
+        expressions: Vec<GroupExpression>,
+    },
+    Not {
+        expression: Box<GroupExpression>,
+    },
+    UnaryComparisonOperator {
+        target: GroupComparisonTarget,
+        operator: UnaryComparisonOperator,
+    },
+    BinaryComparisonOperator {
+        target: GroupComparisonTarget,
+        operator: String,
+        value: GroupComparisonValue,
+    },
+}
+// ANCHOR_END: GroupExpression
+
+// ANCHOR: GroupComparisonTarget
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Aggregate Comparison Target")]
+pub enum GroupComparisonTarget {
+    Aggregate { aggregate: Aggregate },
+}
+// ANCHOR_END: GroupComparisonTarget
+
+// ANCHOR: GroupComparisonValue
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Aggregate Comparison Value")]
+pub enum GroupComparisonValue {
+    Scalar { value: serde_json::Value },
+    Variable { name: String },
+}
+// ANCHOR_END: GroupComparisonValue
+
+// ANCHOR: Dimension
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[skip_serializing_none]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Dimension")]
+pub enum Dimension {
+    Column {
+        /// The name of the column
+        column_name: String,
+        /// Path to a nested field within an object column
+        field_path: Option<Vec<String>>,
+    },
```

Comment:
Could we add a relationship path, so we could support grouping by object-related fields? (As discussed [here](https://github.com/hasura/ndc-spec/pull/148#discussion_r1646814189))

Reply:
Done, thanks!